clang/include/clang/CConv/ConstraintsGraph.h

```diff
@@ -12,125 +12,269 @@
 #ifndef _CONSTRAINTSGRAPH_H
 #define _CONSTRAINTSGRAPH_H
 
-#include <boost/graph/graph_traits.hpp>
-#include <boost/graph/adjacency_list.hpp>
 #include "clang/CConv/Constraints.h"
+#include "llvm/ADT/DirectedGraph.h"
+#include "llvm/Support/GraphWriter.h"
+#include "llvm/ADT/GraphTraits.h"
+#include "llvm/ADT/BreadthFirstIterator.h"
 
-using namespace std;
+template<class DataType> struct DataEdge;
 
-template<class G>
-class BaseGraph {
+template<typename DataType, typename EdgeType = DataEdge<DataType>>
+class DataNode : public llvm::DGNode<DataNode<DataType, EdgeType>, EdgeType> {
```

Comment:
Why do you call it `DataNode` ?

Reply:
I used to have `BaseNode` class that was extended to contain the `Data` field, but I found a better way to write this that avoid the extra class. Could change the name back to `BaseNode` or something else.